src/IO.ts

```diff
@@ -1,6 +1,74 @@
 /**
- * @file `IO<A>` represents a synchronous computation that yields a value of type `A` and **never fails**.
- * If you want to represent a synchronous computation that may fail, please see `IOEither`.
+ * @file `IO<A>` represents a synchronous computation that yields a value of type `A` and **never fails**.  If you want
+ * to represent a synchronous computation that may fail, please see `IOEither`.  IO actions are terminated by calling
+ * their `run()` method that executes the computation and returns the result.
```

Comment:
Maybe a note regarding `run()`  should be called sparingly or even only once at the "end of the program" in order to maintain the referential transparency and composability advantages?

Reply:
I think it would also make sense to update this to say "… represents a synchronous effectful computation", because `IO` lets us describe side-effects and non-deterministic computations.